operator/cmd/mesh/install.go

```diff
@@ -196,6 +196,13 @@ func Install(rootArgs *RootArgs, iArgs *InstallArgs, logOpts *log.Options, stdOu
 	isDefaultInstallation := rev == "" && iop.Spec.Components.Pilot != nil && iop.Spec.Components.Pilot.Enabled.Value
 	operatorManageWebhooks := operatorManageWebhooks(iop)
 
+	if !operatorManageWebhooks && isDefaultInstallation {
```

Comment:
Instead of deleting the conflicting webhook before a new upgrade, why not deactivate it after a failed upgrade since the conflict is caused by the upgrade failure? This will also prevent webhook overlapping if you don't upgrade it again.

Reply:
I think deactivating the webhook will lead to sidecar injection downtime again, but moving deleting conflicting webhook to error might be a good idea so that we can unblock the upgrade right away.

Reply:
just coming back to this, I think removing it after failure will be much better approach

Reply:
Sorry for the delay. +1 for performing the removal after failure. However, IIRC, something may break if we do deletion instead of deactivation. It might be the istioctl tag command, but I'm not quite sure. Need @Monkeyanator to confirm this. Otherwise LGTM.

Reply:
Thanks for coming back, this should be safe now, it is restoring this [pr](https://github.com/istio/istio/pull/34939) 
But now it will check and only remove the default tag webhook when the default revision webhook is not deactivated

I could also create some tests in pkg/tag to verify this scenario.